src/algorithms/backtracking.py

```diff
@@ -0,0 +1,180 @@
+	 		if used:
+	 			return True
+	 	return False 
+
+	def is_used_in_row_of_box(self, grid, row, box_start_row, box_start_col, num):
+		""" This method return a boolean that means if the 'num' number is already 
+		used in the row of a box.
+
+		int[9][9] grid -- the grid with the current sudoku values.
+		int row -- The current row of the box where to check the number.
+		int box_start_row -- row number in the grid where the box 3x3 starts.
+		int box_start_col -- column number in the grid where the box 3x3 starts
+		int num -- number to eval in the box.
+		return boolean -- return True when is already used in the row of a box3x3 or False
+			when is not used.
+		"""
+		for col in range(3):
+	 		if grid[row + box_start_row][col + box_start_col] == num:
+	 			return True
+	 	return False
+
+	def is_safe(self, grid, row, col, num):
+		""" Function that eval a number *num* provided whether this is valid or not 
+		in the provided *position [row,col]*.
+
+		Keyword arguments
+		int[9][9] grid-- the grid with the current sudoku values.
+		int row -- row number in the grid.
+		int col -- column number in the grid.
+		int num -- number to eval in the sudoku grid.
+		return boolean -- return True when the number is valid or False in other case
+		"""
+	 	return  (not self.is_used_in_row(grid,row,num) 
+	 			and not self.is_used_in_col(grid, col, num) 
+	 			and not self.is_used_in_box(grid, row - row % 3, col - col % 3, num))
+
+	def solve_backtraking(self, grid):
+		""" Function that solve a sudoku game with the backtracking algorithm
+		int[9][9] grid -- the grid with the current sudoku values.
+		return boolean -- return True the Grid was solved and False when there is 
+			not solution.
+		"""
+		row = 0
+		col = 0
+		position = self.find_unassigned_location(grid, row, col)
+		if (position is None):
+			return True
+		row = position[0]
+		col = position[1]
+		for number in range(1,10):
+			filled = self.fill_cells(grid, row, col, number)
+			if filled:
+				return True
+	 	return False
+
+	def fill_cells(self, grid, row, col, number):
+		""" This method fill a cell from the 'grid' in the position [row,col] with
+		the 'number' number if the number accomplish the rules in the position and
+		call to solve_backtraking in order to fill the remaining cells.
+
+		int[9][9] grid -- the grid with the current sudoku values.
+		int row --  the current start row position of the cell.
+		int col -- the current start column position of the cell.
+		int number -- the current start number to fill in the cell.
+		return boolean -- return True the Grid was solved and False when there is 
+			not solution.
+		"""
+		if (self.is_safe(grid, row, col, number)):
+			grid[row][col] = number
+			if(self.solve_backtraking(grid)):
+				return True
+			grid[row][col] = self.UNASSIGNED
+
+
+	def convert_to_grid(self, grid_string):
+		""" Function that convert the provided string to a grid 9x9
+
+		string grid_string -- the string to convert
```

Comment:
the string to convert, how is the format of this string, we need more information or an example that makes this clear